botnim/kb/manager.py

```diff
@@ -0,0 +1,179 @@
+        3. Regular files (files: glob_pattern)
+        
+        Args:
+            context_config: Configuration dictionary specifying the source type and location
+            
+        Returns:
+            List of documents in the format required by the vector store
+        """
+        documents = []
+        
+        # Handle spreadsheet source
+        if context_config.get('type') == 'spreadsheet' and 'source' in context_config:
+            from .download_sources import download_and_convert_spreadsheet
+            split_path = (
+                self.config_dir / context_config['split'] 
+                if 'split' in context_config 
+                else self.config_dir / f"{context_config['name']}_split"
+            )
+            documents.extend([
+                (self._add_environment_suffix(filename), file_obj, content_type)
+                for filename, file_obj, content_type 
+                in download_and_convert_spreadsheet(
+                    context_config['source'], 
+                    split_path, 
+                    context_config['name']
+                )
+            ])
+        
+        # Handle split files
+        elif 'split' in context_config:
+            documents.extend(self._process_split_file(context_config))
+        
+        # Handle regular files
+        elif 'files' in context_config:
+            documents.extend(self._process_files(context_config['files']))
+        
+        return documents
+
+    def _process_directory(self, dir_path: Path) -> List[Tuple[str, str, str]]:
+        """Process all markdown files in a directory"""
+        documents = []
+        for file_path in sorted(dir_path.glob('*.md')):
+            if file_path.read_text().strip():
+                env_filename = self._add_environment_suffix(file_path.name)
+                documents.append((
+                    env_filename,
+                    str(file_path),
+                    'text/markdown'
+                ))
+        return documents
+
+    def setup_contexts(self, contexts: list) -> str:
+        """Set up all contexts and return the vector store ID
+        
+        Args:
+            contexts: List of context configurations
+            
+        Returns:
+            str: ID of the created vector store
+        """
+        if not contexts:
+            return None
+        
+        # Create vector store using the first context's name
+        vector_store_id = self.create_vector_store(contexts[0])
+        
+        # Collect and upload documents from all contexts
+        all_documents = []
+        for context in contexts:
+            documents = self.collect_documents(context)
+            if documents:
+                all_documents.extend(documents)
+            else:
+                logger.warning(f"No documents found for context: {context.get('name', 'unnamed')}")
+        
+        if all_documents:
+            self.kb_backend.upload_documents(vector_store_id, all_documents)
```

Comment:
this method would get the list of lists of documents and the configuration of the vector_stores and return the tools/tools_resources.

Reply:
done.